### PR TITLE
Add and list student prizes

### PR DESCRIPTION
**Pull request by ghost**
### Two extra endpoints that manages specific student's prizes have been added.  We can now:

1. List all prizes of a student using `GET api/students/{id}/prizes/`
2. Add new prize using `POST api/students/{id}/prizes/`

Please review and run this code on your machines and let me know if you have any objections.

### CODE REVIEW

File: users/resources.py [hunk 1]
```diff
@@ -104,6 +104,33 @@ def delete(self, request, pk):
...
 
 
+class StudentPrizesResource(APIView):
+    serializer_class = PrizeSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get(self, request, pk):
+        user_id = request.user.id
+
+        if not has_user_access_to_student(user_id, pk):
```
Comment (MateuszBelczowski) [inline users/resources.py]: It's not always necessary to declare variable if we're using it only at one place. As an alternative we could write it as

```suggestion
        if not has_user_access_to_student(request.user.id, pk):
```

but I'll leave the decision up to you

File: users/resources.py [hunk 2]
```diff
@@ -104,6 +104,33 @@ def delete(self, request, pk):
...
+            raise PermissionDenied
+
+        student = Student.objects.get(pk=pk)
+        prizes = Prize.objects.filter(student=student)
```
Comment (MateuszBelczowski) [inline users/resources.py]: It's not necessary to retrieve student object to later use it in the filtering. Filtering can be done directly using the id value
```suggestion
        prizes = Prize.objects.filter(student=pk)
```

or 

```suggestion
        prizes = Prize.objects.filter(student_id=pk)
```

I believe both options should work

File: users/serializers.py [hunk 3]
```diff
@@ -40,7 +40,9 @@ class PrizeSerializer(serializers.Serializer):
     value = serializers.IntegerField()
 
     def create(self, validated_data):
-        return Prize.objects.create(**validated_data)
+        student_data = validated_data.pop('student')
+        student = Student.objects.get(pk=student_data['id'])
```
Comment (MateuszBelczowski) [inline users/serializers.py]: I think the same as above applies here. If I recall correctly you can use id and don't need to retrieve the whole object.

File: users/resources.py [hunk 4]
```diff
@@ -104,6 +104,33 @@ def delete(self, request, pk):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+class StudentPrizesResource(APIView):
```
Comment (MateuszBelczowski) [inline users/resources.py]: I think that we should keep names consistent - other views are named e.g. `PointResource` and not `StudentPointsResource` (so beside the `student` prefix, please also note single vs plural). For now I'd stick to this convention, we can always adjust later

**Review by MateuszBelczowski — COMMENTED** [review]
Overall, it looks good, but left a few minor comments. There are ways to simplify this code, e.g.

- write custom permission class to check if user has access to student
- we don't always need to override full methods (like `get`, `post`)

but we're not doing so in other places, so I'd stick to this approach for now and once we get it working, we can refactor it a little bit. Please also note that we're not using pagination here - not sure if that's expected or not

**Review by miniMax18 — APPROVED** [review]

**Pull request merged**